runestone/assess/js/mchoice.js

```diff
@@ -284,64 +273,109 @@ MultipleChoice.prototype.randomizeAnswers = function () {
     }
 };
 
-/*====================================
-=== Repopulation from localstorage ===
-====================================*/
+/*===================================
+=== Checking/loading from storage ===
+===================================*/
 
-MultipleChoice.prototype.restoreLocalAnswers = function () {         // Handles local storage
-    if (this.multipleanswers) {
-        this.restoreMultipleSelect();
+MultipleChoice.prototype.checkServer = function () {
+    // Check if the server has stored answer
+    if (this.useRunestoneServices) {
+        var data = {};
+        data.div_id = this.divid;
+        data.course = eBookConfig.course;
+        data.event = "mChoice";
+        jQuery.get(eBookConfig.ajaxURL + "getAssessResults", data, this.repopulateFromStorage.bind(this)).error(this.checkLocalStorage.bind(this));
     } else {
-        this.restoreRadio();
+        this.checkLocalStorage();   // just go right to local storage
     }
-};
 
-MultipleChoice.prototype.restoreMultipleSelect = function () {
-    // This function repopulates MCMA questions with a user"s previous answers,
-    // which were stored into local storage.
-    var len = localStorage.length;
-    if (len > 0) {
+};
 
-        var ex = localStorage.getItem(eBookConfig.email + ":" + this.divid);
-        if (ex !== null) {
-            var arr = ex.split(";");
-            var answers = arr[0].split(",");
+MultipleChoice.prototype.repopulateFromStorage = function (data, status, whatever) {
+    if (data !== "") {
+        var dataEval = JSON.parse(data);
+        if (this.shouldUseServer(dataEval)) {
+            var answers;
+            if (this.multipleanswers) {
+                answers = dataEval.answer.split(",");
+            } else {
+                answers = [dataEval.answer.charCodeAt(0) - 97];   // Get index for lowercase letter
```

Comment:
It also appears that answer and correct are switched around, as in my test answer is 'T' and correct is 'c'  this is switched around from how they are in the database.